frontend/packages/yki/src/tests/cypress/integration/public_registration_page.spec.ts

```diff
@@ -8,4 +9,47 @@ describe('PublicRegistrationPage', () => {
   it('is visible', () => {
     onPublicRegistrationPage.isVisible();
   });
+
+  describe('allows filtering exams', () => {
+    it('but filter criteria must be selected first', () => {
+      onPublicRegistrationPage.showResults();
+      const dialogHeading = 'Valitse tutkinnon kieli ja taitotaso';
+      findDialogByText(dialogHeading).should('be.visible');
+      findDialogByText(dialogHeading)
+        .findByText('takaisin', { exact: false })
+        .click();
+      cy.findByRole('dialog').should('not.exist');
+    });
+
+    it('all results are available initially', () => {
+      onPublicRegistrationPage.expectResultsCount(10);
+      onPublicRegistrationPage.selectExamLanguage('kaikki kielet');
+      onPublicRegistrationPage.selectExamLevel('kaikki tasot');
+      onPublicRegistrationPage.expectResultsCount(10);
+      onPublicRegistrationPage.showResults();
```

Comment:
showResults() kutsuminen täällä tarpeetonta? Ellei tällä sitten ajatus tsekata että dialogia kielen ja tason valinnasta ei enää näytetä jonka vois tän perään sinänsä tarkistaa.

Reply:
Joo, sehän tuossa oli ajatuksena.. dialogin näkyvyyden tsekkaus suoraan oli jotenkin hankalaa.

Reply:
Laittaisko tähän testin loppuun sitten myös tuon yo. kohdan `cy.findByRole('dialog').should('not.exist');` ?

Reply:
Ei kun hups, tuo dialogin näkyvyyden tarkistaminen olikin jo yhtä testikeissiä ylempänä. Taidanpa ottaa tän kokonaan pois.